ambry-store/src/test/java/com.github.ambry.store/IndexTest.java

```diff
@@ -798,6 +798,37 @@ public void getLogSegmentsNotInJournalTest() throws InterruptedException, IOExce
   }
 
   /**
+   * Tests the cases where tokens have to be revalidated on a call to
+   * {@link PersistentIndex#findEntriesSince(FindToken, long)}.
+   * @throws StoreException
+   */
+  @Test
+  public void tokenRevalidationTest() throws StoreException {
```

Comment:
That sounds like an integration test - will give it a shot and see if it can be done deterministically (since hard delete is a thread of its own).